modules/server/src/network/aggregations/index.ts

```diff
@@ -172,3 +168,31 @@ const resolveNumericAggregation = (aggregations: NumericAggregations) => {
 	// TODO: implement
 	throw Error('Not implemented');
 };
+
+const emptyAggregation: NetworkAggregation = { bucket_count: 0, buckets: [] };
+
+export class AggregationAccumulator {
+	totalAgg: any;
+
+	constructor(requestedFields: any) {
+		/*
+		 * seed accumulator with the requested field keys
+		 * this will make it easier to add to using key lookup instead of Array.find
+		 */
+		this.totalAgg = Object.keys(requestedFields).reduce((accumulator: any, field: any) => {
+			return { ...accumulator, [field]: emptyAggregation };
+		}, {});
+	}
+
+	resolve(data, fields) {
+		resolveAggregations({
+			accumulator: this.totalAgg,
+			networkResult: data,
+			requestedAggregationFields: fields,
+		});
+	}
+
+	result() {
+		return this.totalAgg;
+	}
+}
```

Comment:
isolates all the accumulator logic from this module cleanly into an exposed class